include/exadg/structure/spatial_discretization/operators/continuum_mechanics.h

```diff
@@ -30,40 +30,177 @@ namespace ExaDG
 {
 namespace Structure
 {
-template<int dim, typename Number = double>
+template<int dim, typename Number, typename TensorType>
 inline DEAL_II_ALWAYS_INLINE //
-  dealii::Tensor<2, dim, dealii::VectorizedArray<Number>>
-  add_identity(dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> gradient)
+  TensorType
+  add_identity(TensorType tensor)
 {
   for(unsigned int i = 0; i < dim; i++)
-    gradient[i][i] = gradient[i][i] + 1.0;
-  return gradient;
+  {
+    tensor[i][i] = tensor[i][i] + 1.0;
+  }
+
+  return tensor;
 }
 
-template<int dim, typename Number = double>
+template<int dim, typename Number, typename TensorType>
 inline DEAL_II_ALWAYS_INLINE //
-  dealii::Tensor<2, dim, dealii::VectorizedArray<Number>>
-  subtract_identity(dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> gradient)
+  TensorType
+  subtract_identity(TensorType tensor)
 {
   for(unsigned int i = 0; i < dim; i++)
-    gradient[i][i] = gradient[i][i] - 1.0;
-  return gradient;
+  {
+    tensor[i][i] = tensor[i][i] - 1.0;
+  }
+
+  return tensor;
 }
 
+// Create a symmetric tensor from a tensor H plus H^T.
+// Note that we have (H + H^T)^T = H^T + (H^T)^T = H^T + H = H + H^T.
 template<int dim, typename Number>
 inline DEAL_II_ALWAYS_INLINE //
-  dealii::Tensor<2, dim, dealii::VectorizedArray<Number>>
-  get_F(const dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> & H)
+  dealii::SymmetricTensor<2, dim, dealii::VectorizedArray<Number>>
+  compute_H_plus_HT(dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & H)
+{
+  dealii::SymmetricTensor<2, dim, dealii::VectorizedArray<Number>> result;
+
+  for(unsigned int i = 0; i < dim; ++i)
+  {
+    for(unsigned int j = i; j < dim; ++j)
+    {
+      result[i][j] = H[i][j] + H[j][i];
+    }
+  }
+
+  // Debug output.
+  if constexpr(false)
```

Comment:
I can also make a test alternatively that checks all the helper functions in `continuum_mechanics.h` ... that felt a bit silly now as it would do something like
```
tensor dst, src;
// fill tensor with random values
dst = compute_H_times_HT(src);
// compute error
```
If the implementation were less straight-forward, I would do it.
I would say i we ever extend it towards a possibly more optimized variant(?), we go for the test.